features/keychain/api/index.d.ts

```diff
@@ -24,16 +35,67 @@ type PrivateKeys = {
   xpriv: string
 }
 
+type PubKeyWrapper = {
+  publicKey: Buffer
+}
+
+type SignerParam = KeychainSignerParams & { seedId: string }
+
+type Encryptor = {
+  getSodiumKeysFromSeed: (
+    args: KeySource
+  ) => Promise<{ box: PubKeyWrapper; sign: PubKeyWrapper; secret: Buffer }>
+  sign: (args: SignerParam) => Promise<Buffer>
+  signOpen: (args: SignerParam) => Promise<Buffer>
+  signDetached: (args: SignerParam) => Promise<Buffer>
+  verifyDetached: (
+    args: SignerParam & {
+      signature: Buffer
+    }
+  ) => Promise<boolean>
+  encryptSecretBox: (args: SignerParam) => Promise<Buffer>
+  decryptSecretBox: (args: SignerParam) => Promise<Buffer>
+  encryptBox: (args: SignerParam & { toPublicKey: Buffer }) => Promise<Buffer>
+  decryptBox: (
+    args: SignerParam & {
+      fromPublicKey: Buffer
+    }
+  ) => Promise<Buffer>
+  encryptSealedBox: (args: { data: Buffer; toPublicKey: Buffer }) => Promise<Buffer>
+  decryptSealedBox: (args: SignerParam) => Promise<Buffer>
+}
+
+type KeyId = Record<string, string>
+
+type Ed25519Signer = {
+  signBuffer: (args: SignerParam) => Promise<Buffer>
+}
+
+type Secp256k1Signer = {
+  signBuffer: (args: SignerParam) => Promise<Buffer>
+  signSchnorr: (args: SignerParam) => Promise<Buffer>
+}
+
+type CreateSignerArgs = {
+  keyId: KeyId
+}
+
 export interface KeychainApi {
+  syncTx(params: TxSignParams): Promise<Tx>
+  addSeed(seed: Buffer): Promise<SeedId>
+  removeAllSeeds(): void
+  clear(): void
+  clone(): KeychainApi
+  createSodiumEncryptor(keyId: string): Encryptor
+  createEd25519Signer(args: CreateSignerArgs): Ed25519Signer
+  createSecp256k1Signer(args: CreateSignerArgs): Secp256k1Signer
```

Comment:
I see methods here that aren't part of the keychain api. Why is that?
[Here](https://github.com/ExodusMovement/exodus-oss/blob/kichelm/add-missing-typedefs/features/keychain/api/index.js#L3) i see keychain only has:

- exportKey
- arePrivateKeysLocked
- sodium
- ed25519
- secp256k1

Reply:
Oh, good catch, my bad, I was looking at the [keychain module](https://github.com/ExodusMovement/exodus-oss/blob/kichelm/add-missing-typedefs/features/keychain/module/keychain.js#L25), but now I am confused with this [module/index.d.ts](https://github.com/ExodusMovement/exodus-oss/blob/kichelm/add-missing-typedefs/features/keychain/module/index.d.ts#L1) file that we have, why is it exporting the actual modules and not type definitions, am I missing something?

Reply:
cc @kewde

Reply:
I'm not sure actually, we can make it type definitions I suppose

Reply:
we should consider migrating keychain to typescript, it will take away the manual type maintenance burden.